backend/usecase/signup.go

```
@@ -0,0 +1,121 @@
+		return
+	}
+
+	input.Password = string(hashedPassword)
+
+	// the check and insert should be in a transaction to prevent race conditions in case of concurrent requests
+	tx, err := s.db.BeginTx(ctx, pgx.TxOptions{})
+	if err != nil {
+		c.Error(api_error.NewInternalServerError("internal server error", err))
+		return
+	}
+	defer tx.Rollback(ctx)
```

golangci-lint

Error return value of `tx.Rollback` is not checked

(errcheck)